src/store/actions/getNFTAssets.js

```diff
@@ -0,0 +1,38 @@
+export const getNFTAssets = async ({ getters, commit, state }, args) => {
+  const { network, walletId } = args
+  const client = getters.client({
+    network,
+    walletId,
+    asset: 'ETH'
+  })
+
+  const nft = await client.nft.fetch()
+
+  nft.assets.forEach((asset) => {
+    if (state.starredNFTs) {
+      asset['starred'] = state.starredNFTs.filter(
+        (item) =>
+          item.asset_contract.address === asset.asset_contract.address && item.id === asset.id
+      ).length
+        ? true
+        : false
+    } else {
+      asset['starred'] = false
+    }
+  })
+
+  commit('SET_NFT_ASSETS_NUMBER', nft.assets.length)
+
+  let result = nft.assets.reduce(function (r, a) {
+    r[a.collection.name] = r[a.collection.name] || []
```

Comment:
Sorting and grouping should not be in the action and is more of a UI concern.
Rather have all NFTs flat in the structure and sorting/ grouping happening elsewhere. You can add it as a getter if you want - there are similar ones in the getters you can look at